src/libraries/System.Formats.Tar/tests/TarReader/TarReader.File.Tests.cs

```diff
@@ -125,28 +125,33 @@ public class TarReader_File_Tests : TarReader_File_Tests_Base
         public void Read_Archive_LongPath_Over255(TarEntryFormat format, TestTarFormat testFormat) =>
             Read_Archive_LongPath_Over255_Internal(format, testFormat);
 
-        [Fact]
-        public void Read_NodeTarArchives_Successfully()
+        [Theory]
+        [InlineData("node-tar")]
+        [InlineData("tar-rs")]
+        public void Read_TestArchives_Successfully(string subset)
         {
-            string nodeTarPath = Path.Join(Directory.GetCurrentDirectory(), "tar", "node-tar");
+            string nodeTarPath = Path.Join(Directory.GetCurrentDirectory(), "tar", subset);
             foreach (string file in Directory.EnumerateFiles(nodeTarPath, "*.tar", SearchOption.AllDirectories))
             {
                 using FileStream sourceStream = File.Open(file, FileMode.Open, FileAccess.Read, FileShare.Read);
                 using var reader = new TarReader(sourceStream);
 
                 TarEntry? entry = null;
-                while (true)
+                while ((entry = reader.GetNextEntry()) != null)
                 {
-                    Exception ex = Record.Exception(() => entry = reader.GetNextEntry());
-                    Assert.Null(ex);
-
-                    if (entry is null) break;
-
-                    ex = Record.Exception(() => entry.Name);
-                    Assert.Null(ex);
-
-                    ex = Record.Exception(() => entry.Length);
-                    Assert.Null(ex);
+                    Assert.NotNull(entry.Name);
+                    Assert.True(Enum.IsDefined(entry.EntryType));
```

Comment:
This would fail if an entry of one of the node-tar and tar-rs asserts has an entry type that is not in our TarEntryType enum. I checked, and we pass the value directly without validating it is among the predefined enums:

https://github.com/dotnet/runtime/blob/d80d71fa6d805bc9129982fbfc26df951d88cce3/src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Read.cs#L378

Reply:
Those files have known entry types, so this assert is valid for the current set of assets. If in the future, we ever updated assets in those directories from upstream into dotnet/runtime-assets, and those new files have some unknown entry types (very unlikely), we can adjust the code accordingly.